### PR TITLE
Fix requestExistingInputState response

### LINKED ISSUE #47137: [Add2App] The typed characters in a TextField are lost when backgrounding the app with focus on the TextField on Android
Using the new embedding API for Android to embed Flutter in an existing app, it is possible for the TextField to lose its last typed characters when backgrounding the app on Android.

The following simple Flutter module code will create a TextField.

Launch the Flutter module in the existing app
Start typing text in the field
With focus still in the field, background the app using the Home button
Come back to the app. Notice the focus is still in the field and the keyboard is hidden, but the characters disappear.

If you take focus off of the TextField before backgrounding the app, then when coming back, the characters remain.

This is related to #35054, but the fix for that bug either introduced this or revealed it.

The following is the code in the flutter module:
```
// Main app entry
void main() {
   runApp(MyApp());
}

class DefaultEmptyPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        body: Center(
            child: TextField(
              decoration: InputDecoration(
                  labelText: "Enter Text"
              ),
            )
        )
    );
  }
}

class MyApp extends StatelessWidget {
  MyApp({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'My App',
      home: DefaultEmptyPage(),
    );
  }
}
```

### PR DESCRIPTION
**Pull request by dnfield**
## Description

#47177 attempted to fix this issue by more frequently calling `setEditingState`. This resulted in races/contention between the framework and platform side about who owned the editing state value, and caused failures on some Espresso based tests.  I'm adding some tests to make sure that behavior is more clearly exercised in https://github.com/flutter/flutter/pull/47464

## Related Issues

#47177 Was a different attempt to fix this.
#43959 Was where a lot of this code was originally introduced.

Fixes #47137.

## Tests

I added the following tests:

Removed a test that isn't valid - we shouldn't be using setEditingState that way.

Updated a test to use the new property on `TextEditingClient`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [x] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [x] I wrote a migration guide: https://github.com/flutter/website/pull/3462

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by dnfield** [comment]
@hixie - this is an example of a breaking change that will be really hard to make soft.

We have an internal customer who implements the interface I want to change.

I could make it a soft break in a really ugly way by adding the new field to a subclass, checking if I have that subclass, and if not doing something to alert the developer that they're going to lose state when the app is backgrounded on Android.

That seems wrong to me. We'd be writing code that's really ugly and hard to understand, e.g.:

```
if (_currentConnection._client is TextEditingClientWithTextValue) {
  // send state up to platform
} else {
  //  report error with instructions on how to migrate?  Except we'll eventually want to remove the subclass?
}
```

While the migration is actually fairly simple, and the usage of this interface is very limited.

**Comment by dnfield** [comment]
I'm going to make this a soft migration for the internal client by patching them to have the new value as an override with a temporary interface that we'll delete once this rolls.

**Comment by dnfield** [comment]
xref cl/286430188 and b/146568373

File: packages/flutter/lib/src/services/text_input.dart [hunk 1]
```diff
@@ -1047,11 +1057,7 @@ class TextInput {
     // the client ID, as long as we have a _currentConnection.
     if (method == 'TextInputClient.requestExistingInputState') {
       assert(_currentConnection._client != null);
-      _attach(_currentConnection, _currentConfiguration);
-      // This will be null if we've never had a call to [_setEditingState].
-      if (_currentTextEditingValue != null) {
```
Comment (goderbauer) [inline packages/flutter/lib/src/services/text_input.dart]: I wonder if the code would be a little bit clearer leaving this if outside the _attach method. The setEditingState seems like kinda a strange parameter to that method.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/services/text_input.dart (hunk 1 above)
Comment (dnfield) [reply packages/flutter/lib/src/services/text_input.dart]: Sure - it seemed a little awkward to me either way but this way it'll be closer to the actual use case for it.

Comment (dnfield) [reply packages/flutter/lib/src/services/text_input.dart]: Actually redoing this uncovered a potential bug (TextEditingValue  can be null).

**Comment by dnfield** [comment]
This should wait for the internal CL to land first in case it gets grabbed by a roll soon

**Pull request merged**